src/tools/rbd_mirror/pool_watcher/RefreshEntitiesRequest.cc

```diff
@@ -101,7 +101,8 @@ void RefreshEntitiesRequest<I>::handle_mirror_group_list(int r) {
     r = librbd::cls_client::mirror_group_list_finish(&it, &groups);
   }
 
-  if (r < 0 && r != -ENOENT) {
+  // Check for ENOTSUP to handle image mirroring in mixed clusters
+  if (r < 0 && r != -ENOTSUP && r != -ENOENT) {
```

Comment:
Should we log this error at the least ? otherwise we are just ignoring the reason for the empty `&groups` here.

Reply:
It already happens to be logged by `dout(10) << "r=" << r << dendl` at the top of the method.  `librbd::cls_client::mirror_group_list_finish()` can't return ENOENT or ENOTSUP.